src/chigame/users/views.py

```diff
@@ -51,8 +52,16 @@ def get_redirect_url(self):
 @login_required
 def user_list(request):
     users = User.objects.all()
+    friends_table = None
 
-    return render(request, "users/user_detail.html", {"users": users})
+    user_profile = UserProfile.objects.filter(user=request.user).first()
+
+    if user_profile:
+        friends = user_profile.friends.all()
+        friends_table = FriendsTable(friends)  # Passing friends data directly to the table
```

Comment:
This is the intended behaviour because UserProfile has a `friends` attribute while User does not. If the User does not have a corresponding UserProfile (1 - 1), the table will not show up because a User itself cannot have friends.